napari/_qt/tests/test_qt_viewer.py

```diff
@@ -316,3 +317,84 @@ def test_qt_viewer_data_integrity(qtbot, dtype):
     viewer.dims.ndisplay = 2
     datamean = viewer.layers[0].data.mean()
     assert datamean == imean
+
+
+def test_play_axis(qtbot):
+    """Test that play_axis changes the slice on axis 0."""
+
+    viewer = ViewerModel()
+    view = QtViewer(viewer)
+    qtbot.addWidget(view)
+
+    np.random.seed(0)
+    data = np.random.random((10, 10, 15))
+    viewer.add_image(data)
+
+    def increment(e):
+        view.dims.counter += 1
+        # if we don't "enable play" again, view.dims won't request a new frame
+        view.dims._play_ready = True
+
+    view.dims.dims.events.axis.connect(increment)
+
+    axis, interval, nframes = 0, 50, 5
+    view.dims.counter = 0
+    view.dims.play_dim(axis, 1000 / interval)
+    # the 0.5 allows for a little clock jitter...
+    qtbot.wait(interval * (nframes + 0.5))
+    view.dims.stop()
+    # really don't want this to fail due to timing, so we're only making sure it
+    # advanced about the right amount... precise timing will depend on the machine
+    c = int(view.dims.counter)
+    assert c >= nframes - 1
```

Comment:
unfortunately I think this has still failed for one of our osx builds but passed for the other, indicating that there is still some non-determinancy here. Maybe you have to weaken the test even further? Sorry about that, we can always try and improve the testing after this goes in

Reply:
yeah, I'll weaken further.  I don't think it's too big of a compromise at this point.  There may come a day when we need to be able to assert some very specific platform-independent timing, but for now it's probably enough to make sure that the play method advances at least a few frames, and the stop method stops it.  If I give it window of +/- 3 frames, i doubt the tests will ever fail, which is obviously more important for now, and we can leave cross-machine timing precision for another PR (if its even possible).